packages/http-client-java/generator/http-client-generator-mgmt/src/test/java/com/microsoft/typespec/http/client/generator/mgmt/util/FluentUtilsTests.java

```diff
@@ -51,4 +51,12 @@ public void testSplitFlattenedSerializedName() {
         Assertions.assertEquals(Arrays.asList("odata.properties", "virtualNetworkSubnetId"), FluentUtils
             .splitFlattenedSerializedName("odata.properties".replace(".", "\\\\.") + ".virtualNetworkSubnetId"));
     }
+
+    @Test
+    public void testReservedClassName() {
+        // other reserved name should already be handled by CodeNamer
+        final String innerSuffix = "Inner";
+        Assertions.assertEquals("ContextModel",
+            FluentUtils.resourceModelInterfaceClassType("Context" + innerSuffix).getName());
```

Comment:
For myself understanding: curious when will the rename happen?

Seems it only tests `ContextModel` case. Shall we make the function name more specific? `testContextInnerRenameToContextModel`?

Reply:
The **input** parameter of the API has Inner suffix (it is the inner class). We are talking about the **output** of the API (it is the resource interface name, not that class).

Reply:
ok, would be great if we could add this example (in comment) to unit test.
```
Response resource = ..
Response<> response = resource.actionWithResponse(Context.None);
```